src/base/Solution.cpp

```diff
@@ -40,23 +41,33 @@ void Solution::setName(const std::string& name) {
 
 void Solution::setThermo(shared_ptr<ThermoPhase> thermo) {
     m_thermo = thermo;
-    if (m_thermo) {
-        m_thermo->setRoot(shared_from_this());
-    }
 }
 
 void Solution::setKinetics(shared_ptr<Kinetics> kinetics) {
     m_kinetics = kinetics;
-    if (m_kinetics) {
-        m_kinetics->setRoot(shared_from_this());
-    }
 }
 
 void Solution::setTransport(shared_ptr<Transport> transport) {
     m_transport = transport;
-    if (m_transport) {
-        m_transport->setRoot(shared_from_this());
+}
+
+void Solution::addAdjacent(shared_ptr<Solution> adjacent) {
+    if (m_adjacentByName.count(adjacent->name())) {
+        throw CanteraError("Solution::addAdjacent",
+            "Solution '{}' already contains an adjacent phase named '{}'.",
+            name(), adjacent->name());
+    }
+    if (m_thermo && adjacent->thermo()
+        && adjacent->thermo()->nDim() <= m_thermo->nDim())
+    {
+        throw CanteraError("Solution::addAdjacent",
+            "Adjacent phases should have higher dimensionality than the reacting ",
+            "phase.\n'{}' is {}-dimensional while '{}' is {}-dimensional",
+            adjacent->thermo()->name(), adjacent->thermo()->nDim(),
+            m_thermo->name(), m_thermo->nDim());
```

Comment:
Ah - got the context for 'higher-dimensional' was mentioned in a docstring!

Reply:
Are you still looking for an explanation of this somewhere, or does this answer your question?